docker/docker.go

```diff
@@ -210,8 +248,8 @@ func (i *Image) pullReq() (*http.Response, error) {
 		req.Header.Set("Authorization", i.Token)
 	}
 
-	req.Header.Set("Accept", "application/vnd.docker.distribution.manifest.v2+json")
-	req.Header.Set("Accept", "application/vnd.docker.distribution.manifest.list.v2+json")
+	// Prefer v2 manifests
+	req.Header.Set("Accept", "application/vnd.docker.distribution.manifest.v2+json, application/vnd.docker.distribution.manifest.list.v2+json")
```

Comment:
Registries that does not support Manifest v2 Schema 2 will ignore that header and provide Manifest v2 Schema 1.